File: backend/models/notification.model.js
const mongoose = require("mongoose");

const notificationSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      default: null,
    },
    childId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Child",
      default: null,
    },
    doctorId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Doctor",
      default: null,
    },
    title: {
      type: String,
      required: true,
    },
    body: {
      type: String,
      required: true,
    },
    type: {
      type: String,
      enum: [
        "medicine",
        "vaccination",
        "growth",
        "growth_alert",
        "appointment_reminder",
        "general",
        "profile",
        "logout",
        "bracelet",
      ],
      required: true,
    },
    target: {
      type: String,
      enum: ["patient", "doctor"],
      required: true,
    },
    isRead: {
      type: Boolean,
      default: false,
    },
    status: {
      type: String,
      enum: ["pending", "sent", "failed"],
      default: "pending",
    },
    sentAt: {
      type: Date,
      default: null,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Notification", notificationSchema);

File: Sigma_App/lib/models/notification.model.dart
class Notification {
  final String id;
  final String? userId;
  final String? childId;
  final String? doctorId;
  final String title;
  final String body;
  final String type;
  final String target;
  final String? recipientId;
  final String? recipientType;
  final bool isRead;
  final DateTime createdAt;

  Notification({
    required this.id,
    this.userId,
    this.childId,
    this.doctorId,
    required this.title,
    required this.body,
    required this.type,
    required this.target,
    this.recipientId,
    this.recipientType,
    required this.isRead,
    required this.createdAt,
  });

  factory Notification.fromJson(Map<String, dynamic> json) {
    return Notification(
      id: json['_id'],
      userId: json['userId'],
      childId: json['childId'],
      doctorId: json['doctorId'],
      title: json['title'],
      body: json['body'],
      type: json['type'],
      target: json['target'],
      recipientId: json['recipientId'],
      recipientType: json['recipientType'],
      isRead: json['isRead'] ?? false,
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'userId': userId,
      'childId': childId,
      'doctorId': doctorId,
      'title': title,
      'body': body,
      'type': type,
      'target': target,
      'recipientId': recipientId,
      'recipientType': recipientType,
      'isRead': isRead,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}